src/c++/simulator/contact.hpp

```diff
@@ -0,0 +1,44 @@
+﻿#pragma once
+
+#include <simulator/particle.hpp>
+#include <simulator/boundaries.hpp>
+
+#include <vector>
+#include <memory> 
+
+struct ParticleSet
+{
+};
+
+struct ContactPair : public ParticleSet
```

Comment:
I am wondering if this struct is really needed.
If we have a `Domain` that contains all particles and boundaries, we do not need another struct just to keep the `Contacts`. I believe we can perform all pair calculations only having the complete list of particles and boundaries in one place (I assume that this place would be `Domain`).
What do you think?

Reply:
I actually find it useful to have a `ContactPair` information, just to give the code more semantics. At the same time, I agree that the way it is being done may not be ideal. Perhaps just making `using ContactPair = std::pair<T<Particle>, T<Particle>>` would already be enough (must decide how to handle `T`, tough)

Reply:
IMHO, I really don't think this struct is not needed, since we have to contemplate all possible contacts during the calculations. However, if the main goal is to add some semantics, I prefer your suggestion using `std::pair` rather than a new struct just for that.